### ISSUE TITLE
Fix js_of_ocaml integration

### ISSUE THREAD

**Issue by aantron** [aantron/bisect_ppx#334]
The js_of_ocaml integration test has recently started failing with

```
            throw _k_}}}
            ^
0,248,Failure,-2,TypeError: runtime.unix_open is not a function
```

**Comment by aantron** [comment]
This is caused by the fix of https://github.com/ocaml/dune/issues/3622, which led to js_of_ocaml programs linked by Dune calling their `at_exit` hooks. Bisect_ppx's js_of_ocaml integration was (and as of right now, still is) relying on those hooks not being called, as they contain Unix-specific code.

**Comment by pat-rondon** [comment]
Thanks for making this interoperable with JSOO!

There's an additional tweak that's needed for binaries linked with the JSOO runtime library, and which thus have type [`Jsoo_runtime.Error`](https://ocsigen.org/js_of_ocaml/latest/api/js_of_ocaml-compiler/Jsoo_runtime/Error/index.html) registered, as described in https://github.com/ocsigen/js_of_ocaml/issues/1146. Binaries linked with the runtime library will throw an exception of type `Jsoo_runtime.Error` rather than `Failure`. (But binaries _not_ linked with the runtime library continue to throw `Failure`.)

A test demonstrating the issue and potential fix are in #397.
